### PR TITLE
[ray-launcher] enable testing ray launcher on python 3.9

### LINKED ISSUE #1205: [Ray-Plugin] Add support for Python 3.9
Python 3.9 support depends on https://github.com/ray-project/ray/issues/12788

Related to #1062

### PR DESCRIPTION
**Pull request by jieru-hu**
Closes #1205

### CODE REVIEW

File: plugins/hydra_ray_launcher/integration_test_tools/create_integration_test_ami.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def _run_command(command: str) -> str:
     return output
 
 
-@hydra.main(config_name="create_integration_test_ami_config")
+@hydra.main(config_path=None, config_name="create_integration_test_ami_config")
```
Comment (Jasha10) [inline plugins/hydra_ray_launcher/integration_test_tools/create_integration_test_ami.py:32]: This change is unrelated to python3.9 support, right?

Comment (jieru-hu) [reply plugins/hydra_ray_launcher/integration_test_tools/create_integration_test_ami.py:32]: Yes, this is just to fix the warning.

**Review by pixelb — APPROVED** [review]
CI is failing in ssh. I presume that's just a temporary issue

**Comment by jieru-hu** [comment]
thanks for the reviews!

Added some docs for the testing tools, and also added even more logging to the tests so it's a bit easier to debug. All super minor. merging now...
